files/common/usr/bin/get-appliance-version

```diff
@@ -18,6 +18,76 @@
 set -o errexit
 set -o pipefail
 
-zfs get -Hpo value \
+function usage() {
+	echo "Usage: $(basename "$0") [-dMmps]"
+	echo
+	echo "Display the version information for the appliance. By default,"
+	echo "the full version (major, minor, and patch portions) are displayed."
+	echo "The following arguments may be used to refine the version"
+	echo "information:"
+	echo "-d, --debug -- provide the entire version including the jenkins"
+	echo "               information, if it exists."
+	echo "-M, --major -- displays the major version"
+	echo "-m, --minor -- displays the major and minor version"
+	echo "-p, --patch -- displays the major, minor, and patch version"
+	echo "-s, --snapshot -- displays the major, minor, patch, and " \
+		"snapshot information"
+	exit 2
+}
+
+options=$(getopt -l "major,minor,patch,snapshot,debug" -o "Mmpsd" -- "$@")
+eval set -- "$options"
+
+output=$(zfs get -Hpo value \
 	"com.delphix:current-version" \
-	"$(dirname "$(zfs list -Hpo name /)")" | tr -d '\n'
+	"$(dirname "$(zfs list -Hpo name /)")")
+
+opt_major=false
+opt_minor=false
+opt_patch=false
+opt_debug=false
+opt_snapshot=false
+
+if [[ $# -gt 2 ]]; then
+	echo "Error: Only one option may be selected"
+	usage
+fi
+
+while [[ $# -gt 0 ]]; do
+	case "$1" in
+	-d | --debug)
+		opt_debug=true
+		shift
+		;;
+	-M | --major)
+		opt_major=true
+		shift
+		;;
+	-m | --minor)
+		opt_minor=true
+		shift
+		;;
+	-p | --patch)
+		opt_patch=true
+		shift
+		;;
+	-s | --snapshot)
+		opt_snapshot=true
+		shift
+		;;
+	--)
+		shift
+		break
+		;;
+	esac
+done
+
+$opt_debug && echo "$output" | tr -d '\n'
+$opt_major && echo "$output" | cut -d'.' -f1,2 | tr -d '\n'
+$opt_minor && echo "$output" | cut -d'.' -f1-3 | tr -d '\n'
+$opt_patch && echo "$output" | cut -d'-' -f1 | cut -d'.' -f1-4 | tr -d '\n'
+$opt_snapshot && echo "$output" | cut -d'+' -f1
+
+if ! $opt_major && ! $opt_minor && ! $opt_patch && ! $opt_debug; then
```

Comment:
I believe this is broken when the `-s` option is used; e.g.
```
$ ./get-appliance-version -s
6.1.0.0-snapshot.20200330155231164
6.1.0.0
```

Additionally, I think it might be cleaner to rely on setting one of the `opt_` variables to true, to trigger the default behaviour. This way, there's no "special case" for the default handling.

Reply:
Yeah, now that everything is mutually exclusive, I can handle this better in the case statement.